orders.h

```diff
@@ -197,210 +202,220 @@ class SellOrder : public Order {
   public:
-    TurnOrder();
-    ~TurnOrder();
+    TurnOrder(bool repeating = false) : Order(O_TURN), repeating(repeating) {};
+    virtual ~TurnOrder() = default;
+
     bool repeating;
     std::vector<std::string> turnOrders;
 };
 
 class CastOrder : public Order {
   public:
-    CastOrder();
-    ~CastOrder();
+    CastOrder(int spell, int level) : Order(O_CAST), spell(spell), level(level) {};
+    virtual ~CastOrder() = default;
 
     int spell;
     int level;
 };
 
 class CastMindOrder : public CastOrder {
   public:
-    CastMindOrder();
-    ~CastMindOrder();
+    CastMindOrder(int level, UnitId *id = nullptr);
+    virtual ~CastMindOrder();
 
     UnitId *id;
 };
 
 class CastRegionOrder : public CastOrder {
   public:
-    CastRegionOrder();
-    ~CastRegionOrder();
+    CastRegionOrder(int spell, int level, int x = 0, int y = 0, int z = 0);
+    virtual ~CastRegionOrder() = default;
 
     int xloc, yloc, zloc;
 };
 
 class TeleportOrder : public CastRegionOrder {
   public:
-    TeleportOrder();
-    ~TeleportOrder();
+    TeleportOrder(int spell, int level, int x = 0, int y = 0, int z = 0);
+    virtual ~TeleportOrder();
 
     int gate;
     std::list<UnitId *> units;
 };
 
 class CastIntOrder : public CastOrder {
   public:
-    CastIntOrder();
-    ~CastIntOrder();
+    CastIntOrder(int spell, int level, int target);
+    virtual ~CastIntOrder() = default;
 
     int target;
 };
 
 class CastUnitsOrder : public CastOrder {
   public:
-    CastUnitsOrder();
-    ~CastUnitsOrder();
+    CastUnitsOrder(int level);
+    virtual ~CastUnitsOrder();
 
     std::list<UnitId *> units;
 };
 
 class CastTransmuteOrder : public CastOrder {
   public:
-    CastTransmuteOrder();
-    ~CastTransmuteOrder();
+    CastTransmuteOrder(int level, int item, int number);
+    virtual ~CastTransmuteOrder() = default;
 
     int item;
     int number;
 };
 
 class EvictOrder : public Order {
   public:
-    EvictOrder();
-    ~EvictOrder();
+    EvictOrder() : Order(O_EVICT) {};
+    virtual ~EvictOrder();
 
     std::list<UnitId *> targets;
 };
 
 class IdleOrder : public Order {
   public:
-    IdleOrder();
-    ~IdleOrder();
+    IdleOrder() : Order(O_IDLE) {};
+    virtual ~IdleOrder() = default;
 };
 
 class TransportOrder : public Order {
   public:
-    TransportOrder();
-    ~TransportOrder();
+    TransportOrder(int item, int amount, int except, UnitId *target = nullptr)
+        : Order(O_TRANSPORT), item(item), amount(amount), except(except), target(target)
+    {
+    };
+    virtual ~TransportOrder();
 
     int item;
     // amount == -1 means all available at transport time
     // any other amount is also checked at transport time
     int amount;
     int except;
-    int distance;
+    int distance = 0;
 
     enum class TransportPhase {
+        UNDEFINED,
         SHIP_TO_QM,
         INTER_QM_TRANSPORT,
         DISTRIBUTE_FROM_QM
     };
-    TransportPhase phase;
+    TransportPhase phase = TransportPhase::UNDEFINED;
```

Comment:
Actually it would probably be alright to default to SHIP_TO_QM here.  It gets updated to the correct value in all cases in the code as is, but by default most transport orders are going to go from a unit to a QM.

Reply:
I figured that was probably the case, but there's no way to guarantee that, and having a "you forgot to set this variable" case seemed safer if anyone ever touches that code in the future.